lib/modules/manager/asdf/extract.ts

```diff
@@ -0,0 +1,40 @@
+import { logger } from '../../../logger';
+import { isSkipComment } from '../../../util/ignore';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import type { PackageDependency, PackageFile } from '../types';
+
+const upgradeableTooling: Record<
+  string,
+  Pick<PackageDependency, 'depName' | 'datasource' | 'packageName'>
+> = {
+  nodejs: {
+    depName: 'node',
+    datasource: GithubTagsDatasource.id,
+    packageName: 'nodejs/node',
+  },
+};
+
+export function extractPackageFile(content: string): PackageFile {
+  logger.trace('asdf.extractPackageFile()');
+
+  const regex =
+    /^(?<content>(?<toolname>(\w+))\s+(?<version>(\d[\d.]+\d)))([^#]*(#(?<comment>(.*))))?/gm;
+
+  const deps = [...content.matchAll(regex)]
+    .filter((match) => !!match.groups)
+    .map((match) => match.groups!)
+    .filter((groups) => upgradeableTooling[groups['toolname']])
+    .filter((groups) => !isSkipComment((groups['comment'] ?? '').trim()))
+    .map((groups) => {
```

Comment:
use `for...of` loop. please read our best practices guide in docs/development folder

Reply:
sure - I must admit I interpreted your docs here differently ie don't use forEach